src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -291,19 +291,27 @@ protected function doGenerate(array $variables, array $defaults, array $requirem
         }
 
         // add a query string if needed
-        $extra = array_udiff_assoc(array_diff_key($parameters, $variables), $defaults, function ($a, $b) {
+        $extras = array_udiff_assoc(array_diff_key($parameters, $variables), $defaults, function ($a, $b) {
             return $a == $b ? 0 : 1;
         });
 
+        foreach ($extras as $param => $value) {
```

Comment:
let's use array_walk_recursive to handle deeper cast?

Reply:
Hmm I didn't get it to work with array_walk_recursive, help appreciated